src/mysqlprovider.py

```diff
@@ -0,0 +1,121 @@
+#!/usr/bin/env python3
+# Copyright 2021 Canonical Ltd.
+# See LICENSE file for licensing details.
+
+"""MySQLProvider module"""
```

Comment:
I think it would be best to put the provider class in the charmlib. That way, if any other charm wants to implement this interface, they can easily.

Reply:
My understanding is that the charmlib provided by a charm is a facility for Consumers of the charm.